metals/src/main/scala/scala/meta/internal/ansi/AnsiStateMachine.scala

```diff
@@ -4,24 +4,15 @@ package scala.meta.internal.ansi
 
 /**
  * A state machine for filtering out ANSI escape codes from a character stream.
- *
- * In addition to stripping out ANSI escape codes, we also strip out carriage return \r
- * characters from Windows-style "\r\n" line breaks.
  */
 case class AnsiStateMachine(apply: Int => AnsiStateMachine)
 object AnsiStateMachine {
   // Helpful docs on ANSI escape sequences: http://ascii-table.com/ansi-escape-sequences.php
   val Start: AnsiStateMachine = AnsiStateMachine {
     case 27 => AnsiEscape
-    case '\r' => CarriageReturn
     case _ => Print
   }
 
-  val CarriageReturn: AnsiStateMachine = AnsiStateMachine {
```

Comment:
Why dropping `\r` is no longer needed?

Reply:
Process's output is processed using `BufferedReader.readLine` internally so there is no more need to split it by lines by `LineListener`.

Reply:
That's what I was suspecting. Thanks for the explanation!